src/workerd/io/trace.c++

```diff
@@ -106,6 +106,51 @@ void Trace::EmailEventInfo::copyTo(rpc::Trace::EmailEventInfo::Builder builder)
   builder.setRawSize(rawSize);
 }
 
+Trace::HibernatableWebSocketEventInfo::HibernatableWebSocketEventInfo(Type type)
+    : type(type) {}
+
+Trace::HibernatableWebSocketEventInfo::HibernatableWebSocketEventInfo(
+    rpc::Trace::HibernatableWebSocketEventInfo::Reader reader)
+    : type(readFrom(reader)) {}
+
+void Trace::HibernatableWebSocketEventInfo::copyTo(
+    rpc::Trace::HibernatableWebSocketEventInfo::Builder builder) {
+  auto typeBuilder = builder.initType();
+  KJ_SWITCH_ONEOF(type) {
+    KJ_CASE_ONEOF(_, Message) {
+      typeBuilder.setMessage();
```

Comment:
nit: Can I convince you to do `initMessage()` and `initError()` instead? There is something that breaks my brain seeing `setFoo()` with no arguments.